### ISSUE TITLE
version compatibility

### ISSUE THREAD

**Issue by SilvesterYu** [xiaoxiaxusummer/GAI_COMM#2]
Thank you for releasing your amazing work! I would like to set up the environment but am having a lot of version incompatibilities with protobuf. May I know your protobuf version and python version?

**Comment by xiaoxiaxusummer** [comment]
@SilvesterYu 

**Hi, please kindly find below the full package list of my conda environment. 
I used Window operation system for this code project.**

| Name                        | Version           | Build              | Channel       |
|-----------------------------|-------------------|--------------------|---------------|
| absl-py                     | 1.4.0            | pypi_0             | pypi          |
| array-record                | 0.4.0            | pypi_0             | pypi          |
| astunparse                  | 1.6.3            | pypi_0             | pypi          |
| bzip2                       | 1.0.8            | h2466b09_7         | conda-forge   |
| ca-certificates             | 2024.7.4         | h56e8100_0         | conda-forge   |
| cachetools                  | 5.4.0            | pypi_0             | pypi          |
| certifi                     | 2024.7.4         | pypi_0             | pypi          |
| charset-normalizer          | 3.3.2            | pypi_0             | pypi          |
| click                       | 8.1.7            | pypi_0             | pypi          |
| cloudpickle                 | 1.3.0            | pypi_0             | pypi          |
| colorama                    | 0.4.6            | pypi_0             | pypi          |
| contextlib2                 | 21.6.0           | pypi_0             | pypi          |
| decorator                   | 5.1.1            | pypi_0             | pypi          |
| dm-tree                     | 0.1.8            | pypi_0             | pypi          |
| etils                       | 1.3.0            | pypi_0             | pypi          |
| filelock                    | 3.13.1           | pypi_0             | pypi          |
| flatbuffers                 | 24.3.25          | pypi_0             | pypi          |
| fsspec                      | 2024.2.0         | pypi_0             | pypi          |
| gast                        | 0.4.0            | pypi_0             | pypi          |
| google-auth                 | 2.33.0           | pypi_0             | pypi          |
| google-auth-oauthlib        | 0.4.6            | pypi_0             | pypi          |
| google-pasta                | 0.2.0            | pypi_0             | pypi          |
| googleapis-common-protos    | 1.63.1           | pypi_0             | pypi          |
| grpcio                      | 1.65.4           | pypi_0             | pypi          |
| h5py                        | 3.11.0           | pypi_0             | pypi          |
| idna                        | 3.7              | pypi_0             | pypi          |
| importlib-metadata          | 8.2.0            | pypi_0             | pypi          |
| importlib-resources         | 6.4.2            | pypi_0             | pypi          |
| jinja2                      | 3.1.3            | pypi_0             | pypi          |
| keras                       | 2.10.0           | pypi_0             | pypi          |
| keras-preprocessing         | 1.1.2            | pypi_0             | pypi          |
| libclang                    | 18.1.1           | pypi_0             | pypi          |
| libffi                      | 3.4.2            | h8ffe710_5         | conda-forge   |
| libsqlite                   | 3.46.0           | h2466b09_0         | conda-forge   |
| libzlib                     | 1.3.1            | h2466b09_1         | conda-forge   |
| markdown                    | 3.7              | pypi_0             | pypi          |
| markupsafe                  | 2.1.5            | pypi_0             | pypi          |
| ml-collections              | 0.1.1            | pypi_0             | pypi          |
| ml-dtypes                   | 0.2.0            | pypi_0             | pypi          |
| mpmath                      | 1.3.0            | pypi_0             | pypi          |
| networkx                    | 3.0              | pypi_0             | pypi          |
| ninja                       | 1.11.1.1         | pypi_0             | pypi          |
| numpy                       | 1.23.1           | pypi_0             | pypi          |
| oauthlib                    | 3.2.2            | pypi_0             | pypi          |
| openssl                     | 3.3.1            | h2466b09_2         | conda-forge   |
| opt-einsum                  | 3.3.0            | pypi_0             | pypi          |
| packaging                   | 24.1             | pypi_0             | pypi          |
| pillow                      | 10.4.0           | pypi_0             | pypi          |
| pip                         | 24.2             | pyhd8ed1ab_0       | conda-forge   |
| promise                     | 2.3              | pypi_0             | pypi          |
| protobuf                    | 3.19.6           | pypi_0             | pypi          |
| psutil                      | 6.0.0            | pypi_0             | pypi          |
| pyasn1                      | 0.6.0            | pypi_0             | pypi          |
| pyasn1-modules              | 0.4.0            | pypi_0             | pypi          |
| python                      | 3.8.19           | h4de0772_0_cpython | conda-forge   |
| pyyaml                      | 6.0.2            | pypi_0             | pypi          |
| requests                    | 2.32.3           | pypi_0             | pypi          |
| requests-oauthlib           | 2.0.0            | pypi_0             | pypi          |
| rsa                         | 4.9              | pypi_0             | pypi          |
| scipy                       | 1.10.1           | pypi_0             | pypi          |
| setuptools                  | 72.1.0           | pyhd8ed1ab_0       | conda-forge   |
| six                         | 1.16.0           | pypi_0             | pypi          |
| sympy                       | 1.12             | pypi_0             | pypi          |
| tensorboard                 | 2.9.1            | pypi_0             | pypi          |
| tensorboard-data-server     | 0.6.1            | pypi_0             | pypi          |
| tensorboard-plugin-wit      | 1.8.1            | pypi_0             | pypi          |
| tensorflow-datasets         | 4.9.0            | pypi_0             | pypi          |
| tensorflow-estimator        | 2.10.0           | pypi_0             | pypi          |
| tensorflow-gan              | 2.1.0            | pypi_0             | pypi          |
| tensorflow-gpu              | 2.10.0rc1        | pypi_0             | pypi          |
| tensorflow-hub              | 0.16.1           | pypi_0             | pypi          |
| tensorflow-io-gcs-filesystem| 0.31.0           | pypi_0             | pypi          |
| tensorflow-metadata         | 1.13.0           | pypi_0             | pypi          |
| tensorflow-probability      | 0.11.0           | pypi_0             | pypi          |
| termcolor                   | 2.4.0            | pypi_0             | pypi          |
| tf-keras                    | 2.15.0           | pypi_0             | pypi          |
| tfp-nightly                 | 0.11.0.dev20200708| pypi_0            | pypi          |
| tk                          | 8.6.13           | h5226925_1         | conda-forge   |
| toml                        | 0.10.2           | pypi_0             | pypi          |
| torch                       | 2.4.0+cu121      | pypi_0             | pypi          |
| torchaudio                  | 2.4.0            | pypi_0             | pypi          |
| torchvision                 | 0.19.0           | pypi_0             | pypi          |
| tqdm                        | 4.66.5           | pypi_0             | pypi          |
| typing-extensions           | 4.9.0            | pypi_0             | pypi          |
| ucrt                        | 10.0.22621.0     | h57928b3_0         | conda-forge   |
| urllib3                     | 2.2.2            | pypi_0             | pypi          |
| vc                          | 14.3             | h8a93ad2_20        | conda-forge   |
| vc14_runtime                | 14.40.33810      | ha82c5b3_20        | conda-forge   |
| vs2015_runtime              | 14.40.33810      | h3bf8584_20        | conda-forge   |
| werkzeug                    | 3.0.3            | pypi_0             | pypi          |
| wheel                       | 0.44.0           | pyhd8ed1ab_0       | conda-forge   |
| wrapt                       | 1.16.0           | pypi_0             | pypi          |
| xformers                    | 0.0.27.post2     | pypi_0             | pypi          |
| xz                          | 5.2.6            | h8d14728_0         | conda-forge   |
| zipp                        | 3.20.0           | pypi_0             | pypi          |


Hope this solves your problem.
